product-service/helpers/validationsHelper.js

```diff
@@ -0,0 +1,26 @@
+export const isString = value => toString.call(value) === '[object String]';
+
+export const isNumber = value => Number.isFinite(value);
+
+export const isExist = value => value != null;
+
+const validationRules = {
+    'existy': (value) => isExist(value) ? null : 'Not exist',
+    'number': (value) => isNumber(value) ? null : 'Not a number',
+    'string': (value) => isString(value) ? null : 'Not a string',
+};
+
+export const getValidationError = data => {
+    let error = '';
+
+    data.forEach(({ value, property, validations })=> {
+        validations.forEach(rule => {
```

Comment:
You could avoid leading whitespace in the error message. For inspiration (it is not tested :)):
```js
const validationResults = data.flatMap(({ value, property, validations }) =>
   validations.map(rule => ({ property, value, error: validationRules[rule](value) }))
);
const errorResults = validationResults.filter(result => isExist(result.error));
const error = errorResults
    .map(({property, error}) => `${property}: ${error}`)
    .join(' ');
return error || null;
```

Reply:
Actually worked as expected. Thanks!